src/test/java/edu/cornell/kfs/module/ar/document/validation/impl/CuTotalAmountBilledToDateExceedsAwardTotalSuspensionCategoryTest.java

```diff
@@ -0,0 +1,143 @@
+import org.kuali.rice.core.api.util.type.KualiDecimal;
+import org.mockito.Mockito;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import edu.cornell.kfs.module.ar.CuArConstants;
+import edu.cornell.kfs.module.cg.businessobject.AwardExtendedAttribute;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({ContractsGrantsInvoiceDocument.class})
+public class CuTotalAmountBilledToDateExceedsAwardTotalSuspensionCategoryTest {
+    
+    private CuTotalAmountBilledToDateExceedsAwardTotalSuspensionCategory suspensionCategory;
+    private ParameterService parameterService;
+    private ContractsGrantsInvoiceDocument contractsGrantsInvoiceDocument;
+    
+    private static final KualiDecimal STANDARD_AWARD_TOTAL = new KualiDecimal(100);
+    private static final KualiDecimal STANDARD_BUDGET_TOTAL = new KualiDecimal(50);
+
+    @Before
+    public void setUp() throws Exception {
+        suspensionCategory = new CuTotalAmountBilledToDateExceedsAwardTotalSuspensionCategory();
+        parameterService = Mockito.mock(ParameterService.class);
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        suspensionCategory = null;
+        parameterService = null;
+        contractsGrantsInvoiceDocument = null;
+    }
+
+    @Test
+    public void shouldValidateOnBudgetTotalForTrueParam() {
+        configureParameterServiceForSuspensionCheck(Boolean.TRUE);
+        assertTrue(suspensionCategory.shouldValidateOnBudgetTotal());
+    }
+    
+    @Test
+    public void shouldValidateOnBudgetTotalForFalseParam() {
+        configureParameterServiceForSuspensionCheck(Boolean.FALSE);
+        assertFalse(suspensionCategory.shouldValidateOnBudgetTotal());
+    }
+    
+    @Test
+    public void testSuspensionByBudgetAmountLessThanBudget() {
+        configureParameterServiceForSuspensionCheck(Boolean.TRUE);
+        prepareContractsGrantsInvoiceDocument(new KualiDecimal(25), STANDARD_AWARD_TOTAL, STANDARD_BUDGET_TOTAL);
+        assertFalse(suspensionCategory.shouldSuspend(contractsGrantsInvoiceDocument));
+    }
+    
+    @Test
+    public void testSuspensionByBudgetAmountEqualBudget() {
+        configureParameterServiceForSuspensionCheck(Boolean.TRUE);
+        prepareContractsGrantsInvoiceDocument(new KualiDecimal(50), STANDARD_AWARD_TOTAL, STANDARD_BUDGET_TOTAL);
+        assertFalse(suspensionCategory.shouldSuspend(contractsGrantsInvoiceDocument));
+    }
+    
+    @Test
+    public void testSuspensionByBudgetAmountMoreThanBudget() {
+        configureParameterServiceForSuspensionCheck(Boolean.TRUE);
+        prepareContractsGrantsInvoiceDocument(new KualiDecimal(55), STANDARD_AWARD_TOTAL, STANDARD_BUDGET_TOTAL);
+        assertTrue(suspensionCategory.shouldSuspend(contractsGrantsInvoiceDocument));
+    }
+    
+    @Test
+    public void testSuspensionByBudgetAmounNullBudgetTotal() {
```

Comment:
Minor typo in method name; should be "...AmountNull..." instead of "...AmounNull..."